Add setting for worker nodePools to specify --max-pods.

jeremyd opened the pull request:
I found this option useful in a testing cluster.  Normally there is a hard limit of 100 pods per node.  If you're using very small resource limits you can easily fit more pods than 100 on a node.

mumoshu commented:
@jeremyd Thanks for your contribution 👍 
This looks nice. Would you also mind adding a corresponding comment in the cluster.yaml template so that our users can see this feature exists.

jeremyd commented:
Thanks for the feedback @mumoshu.  I added the setting to the cluster.yaml and also changed its type to int32 in the cluster.yaml to make it more user-friendly.

redbaron commented:
This is *really* unpractical to have a flag for each possible kubelet flag. we already have $KUBELET_OPTS there, which oyu can populate from cluster.yaml

jeremyd commented:
Oh, I did not realize that.  Ok thanks!  We don't need this then..
